ambry-rest/src/main/java/com/github/ambry/rest/NettyServer.java

```diff
@@ -72,8 +74,14 @@ public void start() throws InstantiationException {
     long startupBeginTime = System.currentTimeMillis();
     try {
       logger.trace("Starting NettyServer deployment");
-      bossGroup = new NioEventLoopGroup(nettyConfig.nettyServerBossThreadCount);
-      workerGroup = new NioEventLoopGroup(nettyConfig.nettyServerWorkerThreadCount);
+      if (Epoll.isAvailable()) {
+        logger.trace("Using EpollEventLoopGroup in NettyServer.");
+        bossGroup = new EpollEventLoopGroup(nettyConfig.nettyServerBossThreadCount);
+        workerGroup = new EpollEventLoopGroup(nettyConfig.nettyServerWorkerThreadCount);
+      } else {
+        bossGroup = new NioEventLoopGroup(nettyConfig.nettyServerBossThreadCount);
+        workerGroup = new NioEventLoopGroup(nettyConfig.nettyServerWorkerThreadCount);
+      }
```

Comment:
while we are changing the EventLoopGroup to Epoll, we should probably change the ServerSocketChannel to Epoll as well. I suspect that crash was caused by the incompatible EventLoopGroup with ServerSocketChannel.

Reply:
https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/nio/AbstractNioChannel.java#L372 This should be where the check is performed.

Reply:
That is right, the epoll usage should be consistent.